xds/internal/test/xds_integration_test.go

```diff
@@ -51,3 +66,148 @@ type testService struct {
 func (*testService) EmptyCall(context.Context, *testpb.Empty) (*testpb.Empty, error) {
 	return &testpb.Empty{}, nil
 }
+
+var (
+	// Globals corresponding to the single instance of the xDS management server
+	// which is spawned for all the tests in this package.
+	managementServer *e2e.ManagementServer
+	xdsClientNodeID  string
+)
+
+func init() {
+	// The management server is started and stopped from TestMain(), but the
+	// leakcheck runs after every individual test. So, we need to skip the
+	// goroutine which spawns the management server and is blocked on the call
+	// to `Serve()`.
+	leakcheck.RegisterIgnoreGoroutine("e2e.StartManagementServer")
+}
+
+// TestMain sets up an xDS management server, runs all tests, and stops the
+// management server.
+func TestMain(m *testing.M) {
+	cancel, err := setupManagementServer()
+	if err != nil {
+		cancel()
```

Comment:
Are we supposed to call `cancel` on error?  Typically not, and if it's nil, this will panic.

Reply:
I was initially always returning a non nil function from `setupManagementServer`. Changed that.